File: GeeksforGeeks_POTD/June_2026/June_10.py
"""
For every value in arr, checked with binary search itself to check if could be found
TC - O(nlogn), SC - O(1)
"""
class Solution:
    def binarySearchable(self, arr):
        ans=0
        for i in arr:
            l=0
            r=len(arr)-1
            while l<=r:
                m=(l+r)>>1
                if arr[m]==i:
                    ans+=1
                    break
                elif arr[m]>i:
                    r=m-1
                else:
                    l=m+1
        return ans

File: GeeksforGeeks_POTD/June_2026/June_10.java
package GeeksforGeeks_POTD.June_2026;
/*
For every value in arr, checked with binary search itself to check if could be found
TC - O(nlogn), SC - O(1)
*/
public class June_10 {
    public int binarySearchable(int[] arr) {
        int ans =0;
        int n =arr.length;
        for(int i=0;i<n;i++){
            int l =0;
            int r= n-1;
            while (l<=r){
                int mid = l+(r-l)/2;
                if(arr[mid]==arr[i]) {ans++;break;}
                else if (arr[mid]<arr[i]){
                    l=mid+1;
                    if(i<l) break;
                }else{
                    r= mid-1;
                    if(i>r) break;
                }
            }
        }
        return ans;
    }
}
